packages/diagrams/src/index.tsx

```diff
@@ -1,14 +1,16 @@
 import type { NodeRenderer } from '@myst-theme/providers';
 import { useEffect, useId, useState } from 'react';
 import classNames from 'classnames';
+import type { Mermaid } from 'mermaid';
 
-async function parse(id: string, text: string): Promise<string> {
-  const { default: mermaid } = await import('mermaid');
-  return await new Promise<string>((resolve) => {
-    (mermaid as any).render(id, text, (code: any) => {
-      resolve(code);
-    });
-  });
+let _mermaid: Mermaid | undefined = undefined;
```

Comment:
```suggestion
// Follow previous flow of importing mermaid dynamically.
// This is just a refactor, it might not be necessary.
let _mermaid: Mermaid | undefined = undefined;
```